packages/odyssey-react/src/components/TextInput/TextInput.test.tsx

```diff
@@ -40,9 +40,34 @@ describe("TextInput", () => {
   });
 
   it("renders a generated id associating the input and label", () => {
-    render(<TextInput label={label} required />);
+    const { container } = render(<TextInput label={label} />);
+
+    const inputElem = container.querySelector("input");
+    const labelElem = container.querySelector("label");
+    const inputElemId = inputElem?.getAttribute("id");
+    const labelElemForAttrValue = labelElem?.getAttribute("for");
+    expect(inputElem).toBeInstanceOf(HTMLInputElement);
+    expect(labelElem).toBeInstanceOf(HTMLLabelElement);
+    expect(labelElemForAttrValue).toEqual(inputElemId);
+  });
+
+  it("renders an aria-describedby attribute when an error is present", () => {
+    const { container } = render(<TextInput label={label} error="oops" />);
+
+    const inputElem = container.querySelector("input");
+    const ariaDescribedById = inputElem?.getAttribute("aria-describedby");
+    const describedByIdElem = container.querySelector(
+      `[id="${ariaDescribedById}"]`
+    );
+    expect(describedByIdElem).toBeInTheDocument();
+  });
+
+  it("does not render an aria-describedby attribute when no error is present", () => {
+    const { container } = render(<TextInput label={label} />);
 
-    expect(screen.getByRole(textBox, { name: label })).toBeVisible();
```

Comment:
This assertion didn't seem to be testing if a generated id associates the input and label elements.

Reply:
it does! it uses the association between the accessibility role and corresponding label in the DOM. Check react-testing library to learn more about this API